rlp/decode_test.go

```diff
@@ -817,3 +943,35 @@ func unhex(str string) []byte {
 	}
 	return b
 }
+
+// This tests the validity checks for fields with struct tag "optional".
+func TestInvalidOptionalField(t *testing.T) {
+	type (
+		invalid1 struct {
+			A uint `rlp:"optional"`
+			B uint
+		}
+		invalid2 struct {
+			T []uint `rlp:"tail,optional"`
+		}
+		invalid3 struct {
+			T []uint `rlp:"optional,tail"`
+		}
+	)
+	tests := []struct {
+		v   interface{}
+		err string
+	}{
+		{v: new(invalid1), err: `rlp: struct field rlp.invalid1.B needs "optional" tag`},
```

Comment:
are there any more combinations of errors ?

Reply:
No, we didn't have any before and here is the [parseStructTag](https://github.com/vechain/go-ethereum/blob/feat/optional-rlp-tag/rlp/typecache.go#L140) function where all the forbidden combinations are checked

Reply:
does this two additional ones for slices are something we need to check too ?

Reply:
These are already checked [here](https://github.com/vechain/go-ethereum/blob/feat/optional-rlp-tag/rlp/decode_test.go#L506) and [here](https://github.com/vechain/go-ethereum/blob/feat/optional-rlp-tag/rlp/decode_test.go#L511)